package/MDAnalysis/topology/MOL2Parser.py

```diff
@@ -148,7 +156,22 @@ def parse(self, **kwargs):
 
         n_atoms = len(ids)
 
-        masses = guessers.guess_masses(types)
+        validated_elements = []
+        invalid_elements = set()
+        for at in types:
+            try:
```

Comment:
How big of a performance hit do we take here if all the elements are invalid? Would just an `if`/`else` be more effective? (I'm thinking if someone decides to read an entire gaff2 file then you'll end up hitting the `KeyError` every single time?)

Reply:
Probably worth you testing this on your end too, but it looks like an if/else would be better here in cases where you're hitting a lot of unknown atom types.

Here's my benchmark:
```python
import string
import random

def try_catch(var, dicter):
    try:
        return dicter[var]
    except KeyError:
        return ''


def if_else(var, dicter):
    if var in dicter:
        return dicter[var]
    else:
        return ''


alphabet_numbers = {}
for i, letter in enumerate(string.ascii_letters):
    alphabet_numbers[letter] = i

random_ascii = [random.choice(string.ascii_letters) for x in range(100000)]
rand_digit = [random.choice(string.digits) for x in range(100000)]

# ascii pass only - all entries exist in input dictionary - both are ~ 21.5 ms
%timeit -r50 retval = [try_catch(var, alphabet_numbers) for var in rand_ascii]
%timeit -r50 retval = [if_else(var, alphabet_numbers) for var in rand_ascii]

# digits only - all entries do not exist in input dictionary
## ~ 55 ms
%timeit -r50 retval = [try_catch(var, alphabet_numbers) for var in rand_digit]
## ~ 21 ms
%timeit -r50 retval = [if_else(var, alphabet_numbers) for var in rand_digit]
```

Please do test it on your side too (my laptop is what you would call a potato 🙃 )

I'm really doing a "worst case scenario someone is passing a 10k atoms non-sybyl atomtypes file", which is kinda insane, but also I've seen crazier things be done....

Reply:
Resolving conversation as duplicate.